src/main/java/uk/gov/hmcts/ccd/auditlog/AuditInterceptor.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.ccd.auditlog;
+
+import org.springframework.lang.Nullable;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.hmcts.ccd.auditlog.aop.AuditContext;
+import uk.gov.hmcts.ccd.auditlog.aop.AuditContextHolder;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuditInterceptor extends HandlerInterceptorAdapter {
+
+    private final AuditService auditService;
+
+    public AuditInterceptor(AuditService auditService) {
+        this.auditService = auditService;
+    }
+
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler,
+                           @Nullable Exception ex) {
+        if (handler instanceof HandlerMethod && ((HandlerMethod) handler).hasMethodAnnotation(LogAudit.class)) {
```

Comment:
in terms of design I wouldn't pass down web level classes like HttpServletRequest to the AuditService. Ideally AuditService should not be tied to any web class. We could have the interceptor parsing the request, adding all that info to the context, and invoking the AuditService with the context only so the AuditService knows nothing about HttpServletRequest  and HttpServletResponse

Reply:
agree.. don't like passing http semantics